ang/test/civicase/case/actions/directives/case-actions.directive.spec.js

```diff
@@ -1,25 +1,115 @@
 /* eslint-env jasmine */
 
-describe('Action', function () {
-  var $compile, $rootScope;
+(function (_) {
+  describe('case actions', function () {
+    beforeEach(module('civicase', 'civicase.data', 'civicase.templates'));
 
-  beforeEach(module('civicase', 'civicase.templates'));
+    let element, $compile, $rootScope, CaseActionsData;
 
-  beforeEach(inject(function (_$compile_, _$rootScope_) {
-    $compile = _$compile_;
-    $rootScope = _$rootScope_;
-  }));
+    beforeEach(inject(function (_$compile_, _$rootScope_, _CaseActionsData_) {
+      $compile = _$compile_;
+      $rootScope = _$rootScope_;
+      CaseActionsData = _CaseActionsData_;
+    }));
 
-  describe('basic tests', function () {
-    var element;
+    describe('basic tests', () => {
+      beforeEach(() => {
+        compileDirective();
+      });
 
-    beforeEach(function () {
-      element = $compile('<div civicase-case-actions=[]></div>')($rootScope);
-      $rootScope.$digest();
+      it('compiles the case action directive', () => {
+        expect(element.html()).toContain('ng-repeat="action in caseActions');
```

Comment:
just a note about this because I know this was not added by us, but this is not a great way to test the list of actions if you are testing the view. The right way is to check the compiled child nodes after they are generated.

Reply:
TBH it was added by us. And the idea was just to check if the directive is getting complied with correct html. But I think its not necessary. I will remove it.

Reply:
If we want to check the correct HTML we need to test the children, not the parent element. We can have this expectation and then not even render the children elements and this test would pass.

Let's leave it for now but keep an eye for tests like this one when we go fixing tests.